src/components/DocumentStatus/DocumentStatus.tsx

```diff
@@ -10,19 +10,39 @@ interface DocumentStatusProps {
   className?: string;
 }
 
+interface VerificationFragmentData {
+  did: string;
+  location: string;
+  status: string;
+}
+
 export const IssuedBy = ({ verificationStatus }: DocumentStatusProps) => {
-  const dnsFragmentName = "OpenAttestationDnsTxt";
-  const dnsFragment = verificationStatus.find((status) => status.name === dnsFragmentName);
-  const dnsIdentity = dnsFragment?.data?.every((issuer: { status: string }) => issuer.status === "VALID");
-  const domainNames = dnsFragment?.data
-    ?.map((issuer: { location: string }) => issuer.location.toUpperCase())
-    .join(", ");
-  const formattedDomainNames = domainNames?.replace(/,(?=[^,]*$)/, " and"); // regex to find last comma, replace with and
+  const formatIdentifier = (fragment: VerificationFragment<VerificationFragmentData[]>): string | undefined => {
+    switch (fragment.name) {
+      case "OpenAttestationDnsTxtIdentityProof":
+      // using fall through to get both cases
+      case "OpenAttestationDnsDidIdentityProof":
+        const domainNames = fragment.data?.map((issuer) => issuer.location.toUpperCase()).join(", ");
+        return domainNames?.replace(/,(?=[^,]*$)/, " and"); // regex to find last comma, replace with and
+      case "OpenAttestationDidIdentityProof":
+        const didNames = fragment.data?.map((issuer) => issuer.did.toUpperCase()).join(", ");
+        return didNames?.replace(/,(?=[^,]*$)/, " and"); // regex to find last comma, replace with and
```

Comment:
Actually can have a function `joinIssuers` to perform the string[] > string joining since its the same for the 3 cases right. 

Then the use case will be like 

return joinIssuser(fragment.data?.map(issuer)=>issuer.did.toUpperCase())